Source/WebCore/style/StyleAdjuster.cpp

```diff
@@ -421,38 +427,38 @@ void Adjuster::adjust(RenderStyle& style, const RenderStyle* userAgentAppearance
     if (style.display() == DisplayType::Contents)
         adjustDisplayContentsStyle(style);
 
-    if (m_element && (m_element->hasTagName(frameTag) || m_element->hasTagName(framesetTag))) {
+    if (nonPseudoElement() && (nonPseudoElement()->hasTagName(frameTag) || nonPseudoElement()->hasTagName(framesetTag))) {
```

Comment:
```suggestion
    if (nonPseudoElement() && (nonPseudoElement()->hasTagName(frameTag) || nonPseudoElement()->hasTagName(framesetTag))) {
```

I'm not sure we want to adjust the pseudo-element styles based on the associated element. I think a better pattern is:

```
if (auto element = m_styleable.element) {
    if (element->hasTagName(frameTag) || element->hasTagName(framesetTag))) {
}
```

Reply:
I think that's a behaviour change.

The existing code always passed an Element for non-pseudos, and never passed one for pseudos (if I read all the constructors correctly).

The intention is that `nonPseudoElement` returns a pointer in the same circumstances that `m_element` was non-null.

`m_styleable.element` is never null (it's a reference), so this change would do the tag checks on the originating element of a pseudo, which I think is incorrect.

Reply:
Hmm I see, I wonder if we can name `nonPseudoElement` better like `rendererElement` (as an analogy for `renderer->element()`) or something similar. I'm not a huge fan of `nonPseudoElement`

Reply:
RenderElement::nonPseudoElement is the equivalent function we’d call if we had a renderer.